client/asset/dcr/dcr.go

```diff
@@ -2530,69 +2529,72 @@ func determineTxTree(msgTx *wire.MsgTx) int8 {
 // to transactions that are tracked by the wallet, although full node wallets
 // are able to look up non-wallet outputs that are unspent.
-func (dcr *ExchangeWallet) lookupTxOutput(ctx context.Context, txHash *chainhash.Hash, vout uint32) (*wire.TxOut, uint32, bool, error) {
+//
+// If the value of the spent flag is -1, it could not be determined with the SPV
+// wallet if it is spent, and the caller should perform a block filters scan to
+// locate a (mined) spending transaction if needed.
+func (dcr *ExchangeWallet) lookupTxOutput(ctx context.Context, txHash *chainhash.Hash, vout uint32) (txOut *wire.TxOut, confs uint32, spent int8, err error) {
 	// Check for an unspent output.
 	output, err := dcr.wallet.UnspentOutput(ctx, txHash, vout, wire.TxTreeUnknown)
 	if err == nil {
-		return output.TxOut, output.Confirmations, false, nil
+		return output.TxOut, output.Confirmations, 0, nil
 	} else if !errors.Is(err, asset.CoinNotFoundError) {
-		return nil, 0, false, err
+		return nil, 0, 0, err
 	}
 
 	// Check wallet transactions.
 	tx, err := dcr.wallet.GetTransaction(ctx, txHash)
 	if err != nil {
-		return nil, 0, false, err // asset.CoinNotFoundError if not found
+		return nil, 0, 0, err // asset.CoinNotFoundError if not found
 	}
 	msgTx, err := msgTxFromHex(tx.Hex)
 	if err != nil {
-		return nil, 0, false, fmt.Errorf("invalid hex for tx %s: %v", txHash, err)
+		return nil, 0, 0, fmt.Errorf("invalid hex for tx %s: %v", txHash, err)
 	}
 	if int(vout) >= len(msgTx.TxOut) {
-		return nil, 0, false, fmt.Errorf("tx %s has no output at %d", txHash, vout)
+		return nil, 0, 0, fmt.Errorf("tx %s has no output at %d", txHash, vout)
 	}
 
-	txOut := msgTx.TxOut[vout]
-	confs := uint32(tx.Confirmations)
+	txOut = msgTx.TxOut[vout]
+	confs = uint32(tx.Confirmations)
 
 	// We have the requested output. Check if it is spent.
 	if confs == 0 {
 		// Only counts as spent if spent in a mined transaction,
 		// unconfirmed tx outputs can't be spent in a mined tx.
-		return txOut, confs, false, nil
+		return txOut, confs, 0, nil
 	}
 
 	if !dcr.wallet.SpvMode() {
 		// A mined output that is not found by wallet.UnspentOutput
 		// is spent if the wallet is connected to a full node.
 		dcr.log.Debugf("Output %s:%d that was not reported as unspent is considered SPENT, spv mode = false.",
 			txHash, vout)
-		return txOut, confs, true, nil
+		return txOut, confs, 1, nil
 	}
 
-	// For SPV wallets, only consider the output spent if it pays to the
-	// wallet because outputs that don't pay to the wallet may be unspent
-	// but still not found by wallet.UnspentOutput.
-	var outputPaysToWallet bool
-	for _, details := range tx.Details {
-		if details.Vout == vout {
-			outputPaysToWallet = details.Category == wallet.CreditReceive.String()
-			break
-		}
-	}
-	if outputPaysToWallet {
-		dcr.log.Debugf("Output %s:%d was not reported as unspent, pays to the wallet and is considered SPENT.",
```

Comment:
As the comments in the revisions say, this was never happening in the dex use case.  Verified searching lots of logs for this log statement.